src/main/java/strategy/impl/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,14 @@
+package strategy.impl;
+
+import db.FruitStorage;
+import model.FruitTransaction;
+import strategy.OperationHandler;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction transaction) {
+        int oldQuantity = FruitStorage.fruits.get(transaction.getFruit());
+        int newQuantity = oldQuantity - transaction.getQuantity();
```

Comment:
Maybe you should add a check If newQuantity is greater >= 0?

Reply:
✅